src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/HealthCheckFunctions.cs

```diff
@@ -1,63 +1,82 @@
 #pragma warning restore IDE0060
     {
+        var psqlConnectionHealthCheck = await CheckPsqlConnectionHealthy();
+        var fileShareMountHealthCheck = CheckFileShareMountHealth();
+        var healthCheckResponse = new HealthCheckResponse(
+            PsqlConnection: psqlConnectionHealthCheck,
+            FileShareMount: fileShareMountHealthCheck);
+
+        if (healthCheckResponse.Healthy)
+        {
+            return new OkObjectResult(healthCheckResponse);
+        }
+        
+        return new ObjectResult(healthCheckResponse){
+            StatusCode = StatusCodes.Status500InternalServerError
+        };
+    }
+    
+    private async Task<HealthCheckSummary> CheckPsqlConnectionHealthy()
+    {
+        logger.LogInformation("Attempting to test PSQL health");
+
         try
         {
-            var message = $"Found {await publicDataDbContext.DataSets.CountAsync()} datasets.";
-            logger.LogInformation(message);
-            return message;
+            await publicDataDbContext.DataSets.AnyAsync();
+            return new HealthCheckSummary(true);
         }
         catch (Exception e)
         {
-            logger.LogError(e, "Error encountered when querying Data Sets");
-            throw;
+            logger.LogError(e, "Error encountered when testing PSQL connection health");
+            return new HealthCheckSummary(false, e.Message);
         }
     }
     
-    [Function(nameof(CheckForFileShareMount))]
-    public Task CheckForFileShareMount(
-#pragma warning disable IDE0060
-        [HttpTrigger(AuthorizationLevel.Function, "get")] HttpRequestData request)
-#pragma warning restore IDE0060
+    private HealthCheckSummary CheckFileShareMountHealth()
     {
         logger.LogInformation("Attempting to read from file share");
         
         try
         {
-            if (Directory.Exists(parquetFileOptions.Value.BasePath))
-            {
-                logger.LogInformation("Successfully found the file share mount");
-            }
-            else
+            if (Directory.Exists(dataSetVersionPathResolver.BasePath()))
             {
-                logger.LogError("Unable to find the file share mount");
+                return new HealthCheckSummary(true);
             }
+            
+            return new HealthCheckSummary(false, "File Share Mount folder does not exist");
         }
         catch (Exception e)
         {
             logger.LogError(e, "Error encountered when attempting to find the file share mount");
-            throw;
+            return new HealthCheckSummary(false, e.Message);
         }
-        
-        return Task.CompletedTask;
     }
+
+    public record HealthCheckResponse(HealthCheckSummary PsqlConnection, HealthCheckSummary FileShareMount)
+    {
+        public bool Healthy => PsqlConnection.Healthy && FileShareMount.Healthy;
+    };
+
+    public record HealthCheckSummary(bool Healthy, string? UnhealthyReason = null);
```

Comment:
I was thinking this alternative implementation:

```
    public record HealthCheckSummary(string? UnhealthyReason = null)
    {
        public bool Healthy => UnhealthyReason == null;
    }
```

would avoid needing the boolean param where there's a possibility that a summary can be constructed as healthy together with an unhealthy reason which wouldn't make sense

 `new HealthCheckSummary(Healthy: true, UnhealthyReason: "feeling a bit unhealthy");`

This is fairly academic though so I'm happy if you don't want to make any changes.